### PR TITLE
doc: Add information about lua function calls

### PR DESCRIPTION
**Pull request by tjdevries**
I thought it might be nice to include a little info about how you can call functions in Lua, since it's not always obvious in our examples (particularly to newcomers) what is happening with how we're calling functions.

Also happy to add other sections if anyone thinks anything would have been immediately helpful when they first started learning.

### CODE REVIEW

**Comment by clason** [comment]
cf #12369

**Comment by tjdevries** [comment]
Hmmm, yes this does relate to that quite nicely. Maybe a new doc `lua-plugin` or something like that... It could be in this file as well.

Maybe `lua-tutorial`.

**Comment by clason** [comment]
Or a Wiki page? That would be even more prominent (and searchable!)

File: runtime/doc/lua.txt [hunk 1]
```diff
@@ -95,6 +95,66 @@ Note:
...
+    example_func(1)
+    -- ==== Result ====
+    -- A is: 1
+    -- B is: nil
+<
+
+    Additionally, if any extra parameters are passed, they are discarded
+    completely.
+
+In Lua, it is also possible, if only argument is desired to be passed, to call
```
Comment (clason) [inline runtime/doc/lua.txt:134]: `only one argument`

**Review by nhooyr — APPROVED** [review]
Great addition! Especially the docs re using a dictionary as an argument. I found that syntax confusing initially.

File: runtime/doc/lua.txt [hunk 2]
```diff
@@ -95,6 +95,66 @@ Note:
   plugins using shell which will not work with paths containing semicolons it
   is better to not have them in 'runtimepath' at all.
 
+==============================================================================
+Lua Syntax Information                                         *lua-syntax-help*
```
Comment (justinmk) [inline runtime/doc/lua.txt:99]: "help" in a help tag is redundant

**Pull request merged**